File: packages/javascript/lib/string/trim.ts
export function trim(str?: string | null): string | null {
	// Reject anything that is not a string. The previous `&&` meant a truthy non-string
	// slipped through and threw on `.trim()`. An empty string is a valid input and still
	// returns ''.
	if (typeof str !== 'string') {
		return null;
	}

	return str.trim().replace(/\s{2,}/g, ' ');
}

File: packages/python/qsu/string/trim.py
import re
from typing import Optional

def trim(str: Optional[str] = None):
	# Reject anything that is not a string. The previous `and` meant a truthy non-string
	# slipped through and raised an AttributeError on `.strip()`. An empty string is a
	# valid input and still returns ''.
	if not isinstance(str, type('')):
		return None

	return re.sub(r'\s{2,}', ' ', str.strip())
